File: Sort_probs/K번째수.js
function solution(array, commands) {
    let answer = [];

    let tempArray = [];
    let cnt = 0;

    for (let a = 0; a < commands.length; a++) {
        if ((commands[a][1] - commands[a][0]) < 1) {
            answer.push(array[commands[a][0] - 1]);
        } else {
            for (let i = commands[a][0] - 1; i < commands[a][1]; i++) {
                //console.log(array[i]);
                //tempArray.push(array[i]);
                tempArray[cnt] = array[i];
                cnt++;
            }
            cnt = 0;
            tempArray.sort((a, b) => {
                return a - b;
            });

            console.log(tempArray);

            answer.push(tempArray[commands[a][2] - 1]);

        }


        //console.log(tempArray);
        tempArray = [];
    }


    return answer;
}

// 참고적으로 slice를 사용하면 코드를 대폭 줄일 수 있음!!
function solution(array, commands) {

    // console.log(array);
    // console.log(commands);

    var result = [];

    var temp;
    for (var i = 0; i < commands.length; i++) {

        temp = array.slice(commands[i][0] - 1, commands[i][1]).sort(function(a, b) {
            return a - b;
        })

        result.push(temp[commands[i][2] - 1]);

    }

    return result;

}

File: using_cpp/Sorting/k번째수.cpp
#include <string>
#include <vector>
#include <algorithm>


using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    int n = commands.size();
    for(int t = 0 ; t < n; t++){
        int i = commands[t][0];
        int j = commands[t][1];
        int k = commands[t][2];
        vector<int> temp_v;
        for(int a = i-1; a < j; a++){
            temp_v.push_back(array[a]);
        }
        sort(temp_v.begin(), temp_v.end());
        answer.push_back(temp_v[k-1]);
    }
    
    return answer;
}